# Using Random Forest to Predict Income

## Import Packages

In [1]:
def warn(*args, **kwargs):
    pass
import warnings
warnings.warn = warn

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn import tree
from sklearn.ensemble import RandomForestClassifier

## Import Data and Inspect

In [4]:
income_data = pd.read_csv("income.csv", header = 0, delimiter = ", ")

print(income_data.head())
print(income_data.info())

   age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country income  
0          2174             0              40  United-States  <=50K  
1             0             0             

There are 15 columns and 32561 rows in the data.

## Preprocess Data

In [10]:
## Create Data to Analyze
income_data["sex-int"] = income_data["sex"].apply(lambda row: 0 if row == "Male" else 1)
labels = income_data[['income']]
data = income_data[["age", "capital-gain", "capital-loss", "hours-per-week","sex-int"]]


## Split Data into Training and Test Sets

In [11]:
train_data, test_data, train_labels, test_labels = train_test_split(data, labels, random_state = 1)
print(len(train_data))
print(len(test_data))

24420
8141


There are 24420 entries in our test set and 8141 in our training set.

## Fit Random Forest Model

In [12]:
forest = RandomForestClassifier(random_state = 1)
forest.fit(train_data, train_labels)
print("Accuracy Score:")
forest.score(test_data, test_labels)


Accuracy Score:


0.8272939442328953

## Let's Add Another Column to the Model

In [15]:
## Let's Add if Country is US or other
#print(income_data["native-country"].value_counts())
income_data["country-int"] = income_data["native-country"].apply(lambda row: 1 if row == "United-States" else 0)

## Update data and labels
labels = income_data[['income']]
data = income_data[["age", "capital-gain", "capital-loss", "hours-per-week","sex-int", "country-int"]]

## Get test and traning sets
train_data, test_data, train_labels, test_labels = train_test_split(data, labels, random_state = 1)

## Fit the New Model
forest = RandomForestClassifier(random_state = 1)
forest.fit(train_data, train_labels)
print("Accuracy Score:")
forest.score(test_data, test_labels)


Accuracy Score:


0.823731728288908

The addition of the country column didn't impact the score too much, but it made the score go down slightly.